units/spawnhost_start.go

```diff
@@ -44,52 +47,42 @@ func makeSpawnhostStartJob() *spawnhostStartJob {
 	return j
 }
 
+// NewSpawnhostStartJob returns a job to start a stopped spawn host.
 func NewSpawnhostStartJob(h *host.Host, user, ts string) amboy.Job {
 	j := makeSpawnhostStartJob()
 	j.SetID(fmt.Sprintf("%s.%s.%s.%s", spawnhostStartName, user, h.Id, ts))
-	j.HostID = h.Id
-	j.UserID = user
+	j.SetScopes([]string{fmt.Sprintf("%s.%s", spawnHostStatusChangeScopeName, h.Id)})
```

Comment:
Why do we set scopes here but not in the spawnhostModifyJob?

Reply:
Start/stop/terminate are mutually exclusive states, so I wanted to prevent races between them with the scope. Modifying the spawn host metadata (i.e. the spawn host modify job) could also fail if the host's state changed, but it seemed less important to put a scope on it.

I also waffled on setting scopes for the spawn host modify job because it would prevent people from modifying their spawn host's metadata multiple times in quick succession (e.g. add a tag, save the page, extend the expiration, save the page again) and would prevent users from doing other operations like terminating the host until the modification attempt was finished. My reasoning was that if you were modifying the host metadata, it would be important to still allow other overwriting operations like stopping/terminating the host, which would make the host modification irrelevant. If the modification fails because the user changed the host state, it'll show up in the event log, so I figured it was okay to allow multiple modification jobs regardless of the host state.